packages/theia-integration/src/browser/diagram/glsp-diagram-manager.ts

```diff
@@ -137,12 +120,13 @@ export abstract class GLSPDiagramManager extends WidgetOpenHandler<GLSPDiagramWi
             const config = this.getDiagramConfiguration(options);
             const diagramOptions = this.createDiagramOptions(options);
             const diContainer = config.createContainer(diagramOptions);
-            const glpsClientContribution = this.glspClientProvider.getGLSPClientContribution(this.contributionId);
+            const glpsClientContribution = this.diagramServiceProvider.getGLSPClientContribution(this.contributionId);
             if (!glpsClientContribution) {
                 throw new Error(`No glsp client contribution is registered for id: ${this.contributionId}!`);
             }
+            const factory = this.diagramServiceProvider.getDiagramWidgetFactory(this.diagramType);
```

Comment:
Should we check for `factory` be defined and throw a meaningful error if it isn't? This would probably enable more clear error messages without looking at this code for adopters.

Also, should we rename `factory` to `diagramWidgetFactory` for clarity.

Reply:
`getDiagramWidgetFactory` already throws an error if the the factory can not be found.

Reply:
>Also, should we rename factory to diagramWidgetFactory for clarity.
👍🏼